ros2doctor/ros2doctor/api/package.py

```diff
@@ -97,15 +97,18 @@ def compare_versions(result: Result, local_packages: dict, distro_packages: dict
     :param: boolean value determines which output to populate, msgs or report
     :return: list of warning messages
     """
-    missing_req = ''
-    missing_local = ''
+    missing_req = []
+    missing_local = []
     for name, local_ver_str in local_packages.items():
         if not local_ver_str:
-            missing_local += ' ' + name
-            local_ver_str = ''
+            missing_local.append(name)
         latest_ver_str = distro_packages.get(name, '')
         if not latest_ver_str:
-            missing_req += ' ' + name
+            missing_req.append(name)
+
+        if not local_ver_str or not latest_ver_str:
+            continue  # Logging is already done by missing_local/missing_req
+
         local_ver = version.parse(local_ver_str).base_version
         latest_ver = version.parse(latest_ver_str).base_version
```

Comment:
Sorry, I wasn't clear on my previous comment.  I think we should preserve the `try..except` block here so that *other* errors in version parsing are caught and a warning printed.

```suggestion
        try:
            local_ver = version.parse(local_ver_str).base_version
        except version.InvalidVersion:
            doctor_warn(f'Failed to parse local version '{local_ver_str}' of package '{name}')
        try:
            latest_ver = version.parse(latest_ver_str).base_version
        except version.InvalidVersion:
            doctor_warn(f'Failed to parse latest version '{latest_ver_str}' of package '{name}')
```